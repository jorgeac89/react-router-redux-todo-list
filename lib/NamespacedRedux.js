const createNamespacedActionTypes = namespace => {
    return ActionTypes => {
        const namespacedActionTypes = {};
        Object.entries(ActionTypes).forEach(([key, value]) => {
            namespacedActionTypes[key] = `@${namespace}/${value}`;
        });
        return namespacedActionTypes;
    }
}

export { createNamespacedActionTypes } ;