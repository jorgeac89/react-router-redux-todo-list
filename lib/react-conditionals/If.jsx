import React from 'react';
import PropTypes from 'prop-types';

class If extends React.Component {
  render() {
    const { check, children } = this.props;
    return check ? children : null;
  }
}

If.propTypes = {
  check: PropTypes.bool,
  children: PropTypes.node,
};

If.defaultProps = {
  check: false,
  children: null,
};

export default If;
