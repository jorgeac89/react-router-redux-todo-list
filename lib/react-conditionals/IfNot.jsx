import React from 'react';
import PropTypes from 'prop-types';

class IfNot extends React.Component {
  render() {
    const { check, children } = this.props;
    return !check ? children : null;
  }
}

IfNot.propTypes = {
  check: PropTypes.bool,
  children: PropTypes.node,
};

IfNot.defaultProps = {
  check: true,
  children: null,
};

export default IfNot;
