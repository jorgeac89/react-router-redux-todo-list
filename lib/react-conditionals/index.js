import If from './If';
import IfNot from './IfNot';

export { If, IfNot };
