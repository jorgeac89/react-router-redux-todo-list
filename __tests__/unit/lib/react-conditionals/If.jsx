import React from 'react';
import { shallow } from 'enzyme';
import { If } from 'lib/react-conditionals';

describe('If library', () => {
  const reactElement = <div>React element</div>;

  it('renders the provided React element when the condition is true', () => {
    const shallowRender = shallow(<If check>{reactElement}</If>)
    expect(shallowRender).toMatchSnapshot();
  });

  it('does not renders the provided React element when the condition is false', () => {
    const shallowRender = shallow(<If check={false}>{reactElement}</If>)
    expect(shallowRender).toMatchSnapshot();
  });
});