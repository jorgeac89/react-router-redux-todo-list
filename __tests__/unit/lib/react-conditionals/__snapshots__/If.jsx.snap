// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If library does not renders the provided React element when the condition is false 1`] = `""`;

exports[`If library renders the provided React element when the condition is true 1`] = `
<div>
  React element
</div>
`;
